src/mesh.F90

```diff
@@ -290,34 +290,42 @@ pure function mesh_intersects_2d(m, xyz0, xyz1) result(intersects)
 
     ! Check if line intersects left surface -- calculate the intersection point
     ! y
-    yi = y0 + (xm0 - x0) * (y1 - y0) / (x1 - x0)
-    if (yi >= ym0 .and. yi < ym1) then
-      intersects = .true.
-      return
+    if ( (x0 < xm0 .and. x1 > xm0) .or. (x0 > xm0 .and. x1 < xm0) ) then
```

Comment:
Too much whitespace per [our style guide](http://openmc.readthedocs.io/en/latest/devguide/styleguide.html#whitespace-in-expressions)